roaring/roaring.go

```diff
@@ -953,7 +959,8 @@ func (b *Bitmap) WriteTo(w io.Writer) (n int64, err error) {
 	byte8 := make([]byte, 8)
 
 	// Build header before writing individual container blocks.
-	// Metadata for each container is 8+2+2+4 = sizeof(key) + sizeof(containerType)+sizeof(cardinality) + sizeof(file offset)
+	// Metadata for each container is 8+2+2+4 = sizeof(key) + sizeof(type)+sizeof(cardinality) + sizeof(file offset)
+	// Type is stored as 2 bytes, even though it's only got values 1..3.
```

Comment:
yes, I noticed quite a few opportunities for improvement in our roaring format.... TODO(2.0) maybe

Reply:
Yeah. I'm still pondering a hypothetical mixed 16-bit/32-bit container format, I think on the whole it's probably not worth it, but it might be worth thinking about.